### ISSUE TITLE
Migrate P2ArtifactRepositoryLayout to the new resolver

### ISSUE THREAD

**Issue by laeubi** [eclipse-tycho/tycho#1766]
Currently `P2ArtifactRepositoryLayout` uses outdated stuff, we should migrate to the new resolver instead, as mentioned by
@cstamas in the ASF Slack:

- that class is history, if you look about it's hisyory, was made for maven2... am a bit dissapointed these were not cleared up better, but yes, Guillaume is right, for such things (like adding new layout) it is resolver you should go to, and leave these ancient ones from maven: ideally, you should make a fully functional repo in resolver and then it will work not only with maven but also all the tools that integrate resolver
- that's another problem that maven2 did not differentiate local and remote repo, so for start you may want to implement remote p2 repo?
- (usable even if it is effectively local, would use file transport) (bearbeitet) 
- as p2 local repo layout cannot be done AFAIK with current resolver APIs
- layout provider handles layouts https://github.com/apache/maven-resolver/blob/master/maven-resolver-impl/src/main/[…]lipse/aether/internal/impl/DefaultRepositoryLayoutProvider.java
- Currently there is only one layout factory https://github.com/apache/maven-resolver/blob/master/maven-resolver-impl/src/main/[…]eclipse/aether/internal/impl/Maven2RepositoryLayoutFactory.java
- DefaultRepositoryLayoutProvider.java
- <https://github.com/apache/maven-resolver|apache/maven-resolver>apache/maven-resolver | Hinzugefügt von GitHub
- Maven2RepositoryLayoutFactory.java
- <https://github.com/apache/maven-resolver|apache/maven-resolver>apache/maven-resolver | Hinzugefügt von GitHub
- so idea is that provider has factories injected and goes round robin asking then to create layout. As you see maven2 layout will throw if layout is NOT "default" -- and then factories are depleted => failed.
- For p2 case I'd create "p2" layout (does NOT throw if repository.getContentType == "p2" and creates instance
- you might look into connector as well (the module using layout) https://github.com/apache/maven-resolver/tree/master/maven-resolver-connector-basic/src/main/java/org/eclipse/aether/connector/basic
- but it may be completely enough just to create layout for start
- (minor note: while component name for maven2 layout is "maven2" vs contenType="default" this is actually an exception -- as maven2 layout is THE default. For new P2 layout component I'd use component name "p2" AND same for contentType as well)
- in other words, component name is NOT used to "target" layout factory, it is either it creates instance or not (throws)
- reason is that you can create p2-simple and then p2-newgen and both can coexists, and you can "tune" then using priority which kicks in first (which is asked first for layout instance) (bearbeitet) 
- (also, naming a sisu component with name "default" has several other implications -- hance maven2 name for default layout)

**Comment by laeubi** [comment]
Also see https://github.com/OpenNTF/p2-layout-provider
